app/models/work.rb

```diff
@@ -792,10 +792,13 @@ def multipart?
   end
 
   after_save :update_complete_status
+  # Note: this can mark a work complete but it can also mark a complete work
+  # as incomplete if its status has changed
   def update_complete_status
     # self.chapters.posted.count ( not self.number_of_posted_chapter , here be dragons )
-    if self.chapters.posted.count == expected_number_of_chapters
-      Work.where("id = #{self.id}").update_all("complete = true")
+    self.complete = self.chapters.posted.count == expected_number_of_chapters
+    if self.will_save_change_to_attribute?(:complete)
+      Work.where("id = #{self.id}").update_all("complete = #{self.complete}")
```

Comment:
So stupid question, could we change this to a before_save and reduce the number of writes to the database ? ( and make the code look smaller as it doesn't need the will_save_change_to_attribute?